modules/Pharmacogenetics.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env python
+"""This module adds a more granular description of the phenotype observed in the PharmGKB evidence."""
+
+import argparse
+import json
+import logging
+import sys
+from typing import List, Optional
+
+from openai import OpenAI
+import pyspark.sql.functions as f
+from pyspark import SparkFiles
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType, StructField, StringType, ArrayType
+
+from common.evidence import initialize_sparksession, write_evidence_strings
+from common.ontology import add_efo_mapping
+
+def parse_phenotype_with_gpt(genotype_text: str, openai_client: OpenAI, gpt_model: str = "gpt-3.5-turbo-1106") -> Optional[List[str]]:
```

Comment:
The classification we do for stop reasons and the parsing we apply here are both NLP tasks, but different.
For the first one, we directly load the model and run the inference locally.
For the second one, the inference is happening at OpenAI's servers. This function is only an interface to their API.
So yes, this function could be abstracted for other tasks that require a prompt and a desired output format. Since we are not doing this anywhere else, I've kept it like that. Let me know if you want me to move it somewhere else